In [1]:
# Auto update notebook imports
#%load_ext autoreload
# %autoreload 2

# Backtrack to folder source directory if it doesn't already exist in path
import os
import sys

if os.path.basename(os.getcwd()) == "notebooks":
    os.chdir("..")

sys.path.append(os.getcwd())
print(os.getcwd())


/Users/harryhopalot/Documents/GitHub/dssd_Insulate_ukraine


In [2]:
# UV Data https://www.weather-atlas.com/en/ukraine/kiev-weather-september

UV_DATA = {
    "Jan":  1,
    "Feb":  1,
    "Mar":  2,
    "Apr":  4,
    "May":  5,
    "Jun":  5,
    "Jul":  6,
    "Aug":  5,
    "Sep":  4,
    "Oct":  2,
    "Nov":  1,
    "Dec":  1,
}

# Data for sunlight hours
# https://www.weather-atlas.com/en/ukraine/kiev-weather-september
HRS_SUNLIGHT = {
    "Jan": 2.5,
    "Feb": 2.4,
    "Mar": 4.9,
    "Apr": 8.8,
    "May": 9.6,
    "Jun": 10.7,
    "Jul": 10.9,
    "Aug": 11,
    "Sep": 9.2,
    "Oct": 5.4,
    "Nov": 4.2,
    "Dec": 2.9,
}

We will calculate the average UV per hour by averaging over average time for sun exposure multipled by the max average UV for that month.

In [3]:
# Multiply each month UV_DATA by HR_SUNLIGHT to get total UV exposure per month
UV_EXPOSURE = {k: v * HRS_SUNLIGHT[k]/24 for k, v in UV_DATA.items()}

# Take the average of the UV_EXPOSURE to get the average UV exposure over the year at any time dy or night
AVG_UV_EXPOSURE = sum(UV_EXPOSURE.values()) / len(UV_EXPOSURE)

print(f"UV Data: {UV_EXPOSURE}")
print(f"Average UV Exposure: {AVG_UV_EXPOSURE}")

# Plot using plotly the UV exposure per month
import plotly.graph_objects as go
import numpy as np

fig = go.Figure(data=go.Scatter(
    x=list(UV_EXPOSURE.keys()),
    y=list(UV_EXPOSURE.values()),
    mode='markers',
    marker=dict(
        size=16,
        color=np.asarray(list(UV_EXPOSURE.values())),
        colorscale='Viridis',
        showscale=True
    )
))

fig.update_layout(
    title="UV Exposure per Month",
    xaxis_title="Month",
    yaxis_title="UV Exposure",
    font=dict(
        family="Courier New, monospace",
        size=18,
        color="#7f7f7f"
    )
)

fig.show()

UV Data: {'Jan': 0.10416666666666667, 'Feb': 0.09999999999999999, 'Mar': 0.4083333333333334, 'Apr': 1.4666666666666668, 'May': 2.0, 'Jun': 2.2291666666666665, 'Jul': 2.725, 'Aug': 2.2916666666666665, 'Sep': 1.5333333333333332, 'Oct': 0.45, 'Nov': 0.17500000000000002, 'Dec': 0.12083333333333333}
Average UV Exposure: 1.1336805555555556


Assume a south facing window, where the window constantly looks towards the sun. This is a worst case scenario, as the sun will not always be shining directly into the window. We will however use the incident angle of the sun on the horizon to reduce part of the UV.

This will mean our average UV exposure through the year on a south facing window will be 1.13 * 0.78 = 1.04.

The reasoning behind this is as follows. US Sheets such as polytunnels have to deal with perpendicula UV. Windows sit up right creating an angle between a window and the sun. This angle is the incident angle. The incident angle is the angle between the sun and the window. The incident angle is 0 when the sun is directly above the window. The incident angle is 90 when the sun is on the horizon. The UV rays that hit the window and cause destablisation will be reduced by the incident angle. 

We assume that the average day is the day that the sun sits directly above the equator. This is the equinox. If we assume the sun follows as path with based on the following website in summer at midday the angle is 63.0 and in winter at midday the angle is 16.0. https://www.worlddata.info/europe/ukraine/sunset.php

We can take the average of these and assume that the average incident angle is 38. This is the angle that the sun hits the window at midday on the equinox. We will use this to reduce the UV.

![Image of Inident Angle](./figures/sun_and_incident_angle.png)



With this we get to a value of sin(58) = 0.78. This means that the UV is reduced by 22% due to the incident angle.

Values on lifetimes of LDPE UVI Stabilised plastics most similar to the ones we are using are those used on polytunnels. With this in england which has a similar UV Index to Ukriane polytunnel recommendations are 4 years.

- https://www.americover.com/product/clear-greenhouse-plastic-sheeting/

With this I would expect a south facing window to last a minumum of 5 years.



On this file I would like to get the level of UV exposure that causes damage and then work backwards on our UV calculation.